coordinator/client/zookeeper/connection_test.go

```diff
@@ -133,6 +133,182 @@ func TestZkDriver(t *testing.T) {
 	conn.Close()
 }
 
+func TestZkDriver_Multi(t *testing.T) {
+	basePath := "/basePath"
+	tc, err := zklib.StartTestCluster(1, nil, nil)
+	if err != nil {
+		t.Fatalf("could not start test zk cluster: %s", err)
+	}
+	defer os.RemoveAll(tc.Path)
+	defer tc.Stop()
+	time.Sleep(time.Second)
+
+	servers := []string{fmt.Sprintf("127.0.0.1:%d", tc.Servers[0].Port)}
+
+	drv := Driver{}
+	dsnBytes, err := json.Marshal(DSN{Servers: servers, Timeout: time.Second * 15})
+	if err != nil {
+		t.Fatalf("unexpected error creating zk DSN: %s", err)
+	}
+
+	dsn := string(dsnBytes)
+
+	conn, err := drv.GetConnection(dsn, basePath)
+	defer conn.Close()
+	if err != nil {
+		t.Fatal("unexpected error getting connection")
+	}
+
+	conn.CreateDir("/basePath")
+
+	//
+	// Test creating a new node and setting a non-existent node. Should not commit.
+	//
+	testNode0 := &testNodeT{
+		Name: "test0",
+	}
+	testNode1 := &testNodeT{
+		Name: "test1",
+	}
+
+	multi := conn.NewTransaction()
+	multi.Create("/test0", testNode0)
+	multi.Set("/test1", testNode1)
+	if err = multi.Commit(); err == nil {
```

Comment:
I do, but sadly we get a generic error for all transaction failures with this version of go-zookeeper. The up-to-date version breaks it down by transaction item.